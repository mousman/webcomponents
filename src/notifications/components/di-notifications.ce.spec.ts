import { describe, expect, it } from 'vitest'
import { shallowMount } from '@vue/test-utils'

import DiNotifications from './di-notifications.ce.vue'
import DiNotification from './di-notification.vue'
import DiNotificationForm from './di-notification-form.vue'

describe('DiNotifications.vue', () => {
  localStorage.clear()
  localStorage.setItem(
    'di-notifications',
    '{"domaine1":[{"title":"titre2","type":"success","description":"desc2","id":"6dadbf2f-7573-4924-96f9-ad028a597a88"},{"title":"titre1","type":"info","description":"desc1","id":"1bedc1ef-16c7-4224-9007-70468835435d"}],"domaine2":[{"title":"titre3","type":"error","description":"desc3","id":"3a4413cb-5d37-4c75-bf4c-ddcfd5645f8f"}]}',
  )
  it('renders', () => {
    const wrapper = shallowMount(DiNotifications, {
      props: {
        namespace: 'domaine1',
      },
      slots: {
        default: 'Header',
      },
    })
    expect(wrapper.text()).toBe('Headerdomaine1 2 unread')
    const notifications = wrapper.findAllComponents(DiNotification)
    expect(notifications).toHaveLength(2)
    expect(wrapper.findComponent(DiNotificationForm).exists()).toBe(true)
    expect(notifications?.[0].props('notification')).toMatchSnapshot()
    expect(notifications?.[1].props('notification')).toMatchSnapshot()
  })

  it("doesn't show creation form if not wanted", () => {
    const wrapper = shallowMount(DiNotifications, {
      props: {
        namespace: 'domaine1',
        creation: 'false',
      },
    })
    expect(wrapper.findComponent(DiNotificationForm).exists()).toBe(false)
  })

  it('removes a notification', async () => {
    const wrapper = shallowMount(DiNotifications, {
      props: {
        namespace: 'domaine1',
      },
      slots: {
        default: 'Header',
      },
    })
    const notification = wrapper.findComponent(DiNotification)
    expect(wrapper.findAllComponents(DiNotification)).toHaveLength(2)
    notification.vm.$emit('close', '1bedc1ef-16c7-4224-9007-70468835435d')
    await wrapper.vm.$nextTick()
    expect(wrapper.findAllComponents(DiNotification)).toHaveLength(1)
  })
})
