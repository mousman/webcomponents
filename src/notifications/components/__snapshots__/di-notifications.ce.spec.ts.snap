// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DiNotifications.vue > renders 1`] = `
{
  "description": "desc2",
  "id": "6dadbf2f-7573-4924-96f9-ad028a597a88",
  "title": "titre2",
  "type": "success",
}
`;

exports[`DiNotifications.vue > renders 2`] = `
{
  "description": "desc1",
  "id": "1bedc1ef-16c7-4224-9007-70468835435d",
  "title": "titre1",
  "type": "info",
}
`;

exports[`NotificationCore.vue > renders 1`] = `
{
  "description": "desc2",
  "id": "6dadbf2f-7573-4924-96f9-ad028a597a88",
  "title": "titre2",
  "type": "success",
}
`;

exports[`NotificationCore.vue > renders 2`] = `
{
  "description": "desc1",
  "id": "1bedc1ef-16c7-4224-9007-70468835435d",
  "title": "titre1",
  "type": "info",
}
`;
